pkg/backporter/backporter.go

```diff
@@ -0,0 +1,292 @@
+
+  <div class="collapse navbar-collapse" id="navbarSupportedContent">
+    <form class="form-inline my-2 my-lg-0" role="search" action="/getclones" method="get">
+      <input class="form-control mr-sm-2" type="search" placeholder="Bug ID" aria-label="Search" name="ID">
+      <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Find Clones</button>
+    </form>
+  </div>
+</nav>
+<div class="container">
+`
+
+const htmlPageEnd = `
+</div>
+<footer>
+<p class="small">Source code for this page located on <a href="https://github.com/openshift/ci-tools">GitHub</a></p>
+</footer>
+</body>
+</html>
+`
+
+const clonesTemplateConstructor = `
+	<h2 id="bugid"> <a href = "#bugid"> {{.Bug.ID}}: {{.Bug.Summary}} </a> | Status: {{.Bug.Status}} </h2>
+	<p> Target Release: {{ .Bug.TargetRelease }} </p>
+	{{ if .PRs }}
+		<p> GitHub PR: 
+		{{ if .PRs }}
+			{{ range $index, $pr := .PRs }}
+				{{ if $index}}|{{end}} 
+				<a href="{{ $pr.Type.URL }}/{{ $pr.Org}}/{{ $pr.Repo}}/pull/{{ $pr.Num}}">{{ $pr.Org}}/{{ $pr.Repo}}#{{ $pr.Num}}</a>
+			{{ end }}
+		{{ end }}
+		</p>
+	{{ else }}
+		<p> No linked PRs! </p>
+	{{ end }}
+	{{ if ne .Parent.ID .Bug.ID}}
+		<p> Cloned From: <a href = "/getclones?ID={{.Parent.ID}}"> Bug {{.Parent.ID}}: {{.Parent.Summary}}</a> | Status: {{.Parent.Status}}
+	{{ else }}
+		<p> Cloned From: This is the original! </p>
+	{{ end }}
+	<h4 id="clones"> <a href ="#clones"> Clones</a> </h4>
+	<table class="table">
+		<thead>
+			<tr>
+				<th title="Targeted version to release fix" class="info">Target Release</th>
+				<th title="ID of the cloned bug" class="info">Bug ID</th>
+				<th title="Status of the cloned bug" class="info">Status</th>
+				<th title="PR associated with this bug" class="info">PR</th>
+			</tr>
+		</thead>
+		<tbody>
+		{{ if .Clones }}
+			{{ range $clone := .Clones }}
+				<tr>
+					<td style="vertical-align: middle;">{{ $clone.TargetRelease }}</td>
+					<td style="vertical-align: middle;"><a href = "/getclones?ID={{$clone.ID}}">{{ $clone.ID }}</a></td>
+					<td style="vertical-align: middle;">{{ $clone.Status }}</td>
+					<td style="vertical-align: middle;">
+						{{range $index, $pr := $clone.PRs }}
+							{{ if $index}},{{end}}
+							<a href = "{{ $pr.Type.URL }}/{{$pr.Org}}/{{$pr.Repo}}/pull/{{$pr.Num}}" target="_blank"> {{$pr.Org}}/{{$pr.Repo}}#{{$pr.Num}}</a>
+						{{end}}
+					</td>
+				</tr>
+			{{ end }}
+		{{ else }}
+			<tr> <td colspan=4 style="text-align:center;"> No clones found! </td></tr>
+		{{ end }}
+		</tbody>
+	</table>`
+
+var (
+	clonesTemplate = template.Must(template.New("clones").Parse(clonesTemplateConstructor))
+	emptyTemplate  = template.Must(template.New("empty").Parse("{{.}}"))
+)
+
+// HandlerFuncWithErrorReturn allows returning errors to be logged
+type HandlerFuncWithErrorReturn func(http.ResponseWriter, *http.Request) error
+
+type wrapper struct {
+	Bug    *bugzilla.Bug
+	Clones []*bugzilla.Bug
+	Parent *bugzilla.Bug
+	PRs    []bugzilla.ExternalBug
+}
```

Comment:
It's not obvious to me what this `wrapper` is (what does it wrap?). A godoc would be helpful, ideally also describing what the members are.

Reply:
Oh, so it's a "collection of data for clonesTemplate"! Maybe we should call it `clonesTemplateData` or something like that?

Reply:
Yup. I can make that change